$NetBSD: patch-utils_haddock_haddock.cabal,v 1.2 2023/01/21 16:26:23 pho Exp $

Don't assume we always build the threaded RTS.

Pull request:
https://github.com/haskell/haddock/pull/1575

--- utils/haddock/haddock.cabal.orig	2023-01-11 07:37:47.265320221 +0000
+++ utils/haddock/haddock.cabal
@@ -59,12 +59,16 @@ flag in-ghc-tree
   description: Are we in a GHC tree?
   default: False
   manual: True
+flag threaded
+  default: False
 
 executable haddock
   default-language:     Haskell2010
   main-is:              Main.hs
   hs-source-dirs:       driver
-  ghc-options:          -funbox-strict-fields -Wall -Wincomplete-uni-patterns -Wincomplete-record-updates -O2 -threaded
+  ghc-options:          -funbox-strict-fields -Wall -Wincomplete-uni-patterns -Wincomplete-record-updates -O2
+  if flag(threaded)
+    ghc-options: -threaded
 
   -- haddock typically only supports a single GHC major version
   build-depends:
